encord/http/querier.py

```diff
@@ -157,10 +149,15 @@ def _do_request(req, timeouts: tuple):
 
         return res_json.get("response")
 
-    @staticmethod
-    def create_new_session() -> Session:
-        session = Session()
 
-        session.mount("https://", HTTPAdapter(max_retries=Retry(connect=0)))
+@contextmanager
+def create_new_session(max_retries: Optional[int] = None) -> Session:
+    retry_policy = Retry(total=None, connect=max_retries, read=max_retries)
```

Comment:
Can we not add the backoff factor in here as well?

Reply:
Added back in